packages/quantic/force-app/main/default/lwc/quanticNumberInput/quanticNumberInput.js

```diff
@@ -0,0 +1,66 @@
+import { LightningElement, api } from 'lwc';
+
+import min from "@salesforce/label/c.quantic_Min";
+import max from "@salesforce/label/c.quantic_Max";
+import numberInputMinimum from "@salesforce/label/c.quantic_NumberInputMinimum";
+import numberInputMaximum from "@salesforce/label/c.quantic_NumberInputMaximum";
+import apply from "@salesforce/label/c.quantic_Apply";
+import numberInputApply from "@salesforce/label/c.quantic_NumberInputApply";
+
+import {I18nUtils} from 'c/quanticUtils';
+
+
+export default class QuanticNumberInput extends LightningElement {
+
+    /**@type {number} */
+    @api start;
+    /**@type {number} */
+    @api end;
+    /** @type {string} */
+    @api label;
+
+    /** @type {HTMLInputElement} */
+    inputMin;
+    /** @type {HTMLInputElement} */
+    inputMax;
+
+    minSafeInteger = Number.MIN_SAFE_INTEGER;
+    maxSafeInteger = Number.MAX_SAFE_INTEGER;
+
+    labels = {
+        min,
+        max,
+        numberInputMinimum,
+        numberInputMaximum,
+        apply,
+        numberInputApply
+    }
+
+    renderedCallback() {
+        this.inputMin = this.template.querySelector('.numeric__input-min');
+        this.inputMax = this.template.querySelector('.numeric__input-max');
+    }
```

Comment:
I never know what's better to do here if keeping a reference is better or accessing these every time using getters?

```suggestion
    get inputMin() {
      return this.template.querySelector('.numeric__input-min');
    }
    
    get inputMax() {
      return this.template.querySelector('.numeric__input-max');
    }
```

What do you think? I don't think in this case it makes a huge difference, however, if you had some conditional rendering in your template, I think keeping a direct reference to the element might lead to sometimes having a stale reference to an element that is no longer in the page.

Reply:
getters is better